app/assets/javascripts/surveys.js

```diff
@@ -53,6 +53,9 @@
 
       var $surveyLink = $('#take-survey');
       var surveyUrl = survey.url;
+      if (typeof surveyUrl === "function") {
+        surveyUrl = surveyUrl();
+      }
```

Comment:
Realised we didn't need to support functional urls at all (at least for this survey) so we've dropped this entirely.